tests/memories_test.py

```diff
@@ -1602,6 +1600,10 @@ def g(ys, _):
       self.assertIn('S(5)', compiled_text)
       self.assertNotRegex(compiled_text, r"copy-start.*S\(5\)")
       self.assertNotRegex(compiled_text, r"copy-done.*S\(5\)")
+      self.assertRegex(compiled_text, r"dynamic-update-slice-start.*S\(5\)")
```

Comment:
@hawkinsp I don't understand. upstream JAX test only need to pass with upstream JAX test. Not with the latest JAX release.

So I don't understand why you are asking that? Do I miss somethin?

Reply:
We have builds that run JAX at HEAD with latest jaxlib from pypi and this test will break there because the change to make it work in XLA does not exist in jaxlib from pypi. 

So this test on GPU needs to be guarded on `xla_extension_version`. Note not TPU, that already works.

Reply:
And we have tests that do that because we want to support jax >= jaxlib builds so that you don't have to build jaxlib to do local development where changes only touch JAX and not jaxlib.

Reply:
How to figure out which jaxlib release it is for this [XLA PR 17500](https://github.com/openxla/xla/pull/17500) ?

> So this test on GPU needs to be guarded on xla_extension_version. 
Is this needed by the Bazel to run this Jax unit test?

Reply:
Let me patch your CL and get it in. It'll be way faster for me to do it.

Reply:
Thanks!